server/schema/type.js

```diff
@@ -0,0 +1,86 @@
+const pageInfoType = `
+  type PageInfo {
+    currentPage: Int!
+    totalPages: Int!
+    hasNextPage: Boolean!
+    hasPreviousPage: Boolean!
+    totalItems: Int!
+  }
+`;
+
+const bookType = `
+    type Book {
+      id: ID!
+      title: String!
+      description: String
+      published_date: String
+      author: Author
+      average_rating: Float
+      review_count: Int
+    }
+`;
+
+const bookListType = `
+  type PaginatedBooks {
+    books: [Book!]!
```

Comment:
What does `!]!` mean?

Reply:
The `!` inside means the book cannot be null or undefined and outside one means the array cannot be null or undefined either.